src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/InboundEventsIntegrationTest.kt

```diff
@@ -315,6 +319,80 @@ class InboundEventsIntegrationTest : SqsIntegrationTestBase() {
     notifications.isPresent("p@p.com", CancelledCourtBookingPrisonCourtEmail::class, persistedBooking)
   }
 
+  @Test
+  fun `should remove the pre-appointment for a video court booking on receipt of a prisoner appointment cancelled event`() {
+    videoBookingRepository.findAll() hasSize 0
+    notificationRepository.findAll() hasSize 0
+
+    prisonSearchApi().stubGetPrisoner("123456", PENTONVILLE)
+
+    val courtBookingRequest = courtBookingRequest(
+      courtCode = DERBY_JUSTICE_CENTRE,
+      prisonerNumber = "123456",
+      prisonCode = PENTONVILLE,
+      location = pentonvilleLocation,
+      date = tomorrow(),
+      startTime = LocalTime.of(12, 0),
+      endTime = LocalTime.of(12, 30),
+      comments = "integration test court booking comments",
+      appointments = listOf(
+        Appointment(
+          type = AppointmentType.VLB_COURT_PRE,
+          locationKey = pentonvilleLocation.key,
+          date = tomorrow(),
+          startTime = LocalTime.of(10, 45),
+          endTime = LocalTime.of(11, 0),
+        ),
+        Appointment(
+          type = AppointmentType.VLB_COURT_MAIN,
+          locationKey = pentonvilleLocation.key,
+          date = tomorrow(),
+          startTime = LocalTime.of(11, 0),
+          endTime = LocalTime.of(11, 30),
+        ),
+      ),
+    )
+
+    val bookingId = webTestClient.createBooking(courtBookingRequest, COURT_USER)
+    val createdBooking = videoBookingRepository.findById(bookingId).orElseThrow().also { it.statusCode isEqualTo StatusCode.ACTIVE }
+
+    prisonAppointmentRepository.findByVideoBooking(createdBooking) hasSize 2
+
+    inboundEventsListener.onMessage(
+      raw(
+        PrisonerVideoAppointmentCancelledEvent(
+          personReference = PersonReference(listOf(Identifier("NOMS", "123456"))),
+          additionalInformation = AppointmentScheduleInformation(
+            scheduleEventId = 1,
+            scheduledStartTime = tomorrow().atTime(10, 45),
+            scheduledEndTime = null,
+            scheduleEventSubType = "VLB",
+            scheduleEventStatus = "",
+            recordDeleted = true,
+            agencyLocationId = PENTONVILLE,
+          ),
+        ),
+      ),
+    )
+
+    val updatedBooking = videoBookingRepository.findById(bookingId).orElseThrow()
+
+    with(updatedBooking) {
+      statusCode isEqualTo StatusCode.ACTIVE
+      amendedBy isEqualTo SERVICE_USER.username
+      amendedTime isCloseTo LocalDateTime.now()
+    }
+
+    prisonAppointmentRepository.findByVideoBooking(updatedBooking).single().appointmentType isEqualTo "VLB_COURT_MAIN"
```

Comment:
Two appointments before event is called, now there is only the main appointment.